drivers/hmis/app/models/hmis/hud/client.rb

```diff
@@ -163,40 +163,42 @@ def warehouse_url
     "https://#{ENV['FQDN']}/clients/#{id}/from_source"
   end
 
-  def mci_id
-    ac_hmis_mci_ids.to_a.min_by(&:id)&.value
-  end
-
-  private def clientview_url
+  private def clientview_url(mci_id_value = ac_hmis_mci_ids.to_a.min_by(&:id)&.value)
     link_base = HmisExternalApis::AcHmis::Clientview.link_base
     return unless link_base&.present? && mci_id&.present?
 
-    "#{link_base}/ClientInformation/Profile/#{mci_id}?aid=2"
+    "#{link_base}/ClientInformation/Profile/#{mci_id_value}?aid=2"
   end
 
   def external_identifiers
-    external_identifiers = {
-      client_id: {
+    external_identifiers = [
+      {
+        type: :client_id,
         id: id,
         label: 'HMIS ID',
       },
-      personal_id: {
+      {
+        type: :personal_id,
         id: personal_id,
         label: 'Personal ID',
       },
-      warehouse_id: {
+      {
+        type: :warehouse_id,
         id: warehouse_id,
         url: warehouse_url,
         label: 'Warehouse ID',
       },
-    }
+    ]
 
     if HmisExternalApis::AcHmis::Mci.enabled?
-      external_identifiers[:mci_id] = {
-        id: mci_id,
-        url: clientview_url,
-        label: 'MCI ID',
-      }
+      ac_hmis_mci_ids.to_a.each do |mci_id|
+        external_identifiers << {
+          type: ['mci_id', mci_id.id].join('_'),
+          id: mci_id.value,
+          url: clientview_url(mci_id.value),
+          label: 'MCI ID',
+        }
+      end
```

Comment:
We should preserve the old behavior whereby if the client has _no_ MCI IDs, an ExternalIdentifier for MCI ID still resolves with an empty `value`. That's necessary because we want to keep showing `MCI ID: no value` in the UI in those instances.